test/parallel/test-fs-chmod-mask.js

```diff
@@ -89,6 +90,34 @@ function test(mode, asString) {
     fs.lchmodSync(link, input);
     assert.strictEqual(fs.lstatSync(link).mode & 0o777, mode);
   }
+
+  // Test that `lchmod` works on readonly (0x444) files.
+  if (fs.lchmod) {
+    const link = path.join(tmpdir.path, `lchmod-read-only-src-${suffix}`);
+    const file = path.join(tmpdir.path, `lchmod-read-only-dest-${suffix}`);
+    fs.writeFileSync(file, 'test', { encoding: 'utf-8', mode: 0o444 });
+    fs.symlinkSync(file, link);
+
+    fs.lchmod(link, input, common.mustCall((err) => {
+      assert.ifError(err);
+      assert.strictEqual(
+        fs.lstatSync(link).mode & kMaskOnlyRead, kMaskOnlyRead
+      );
+    }));
+  }
+
+  // Test that `lchmodSync` works on readonly (0x444) files.
+  if (fs.lchmodSync) {
+    const link = path.join(tmpdir.path, `lchmodSync-read-only-src-${suffix}`);
+    const file = path.join(tmpdir.path, `lchmodSync-read-only-dest-${suffix}`);
+    fs.writeFileSync(file, 'test', { encoding: 'utf-8', mode: 0o444 });
+    fs.symlinkSync(file, link);
+
+    for (const _mode of [...Array(1..mode).keys()]) {
```

Comment:
I am not sure I understand the purpose of this loop here. Could you explain?

Reply:
Also, `1..mode`? I read that as the floating point number `1.0` having a `mode` property. I don't follow.

Reply:
Refer https://github.com/nodejs/node/pull/23887#discussion_r228645719